Source/Core/VideoCommon/GraphicsModSystem/Runtime/Actions/ScaleAction.cpp

```diff
@@ -30,30 +30,30 @@ ScaleAction::ScaleAction(Common::Vec3 scale) : m_scale(scale)
 {
 }
 
-void ScaleAction::OnEFB(bool*, u32 texture_width, u32 texture_height, u32* scaled_width,
-                        u32* scaled_height)
+void ScaleAction::OnEFB(const GraphicsModActionData::Input::EFB& input,
+                        GraphicsModActionData::Output::EFB* output)
 {
-  if (scaled_width && m_scale.x > 0)
-    *scaled_width = texture_width * m_scale.x;
+  if (output && m_scale.x > 0)
+    output->scaled_width = input.texture_width * m_scale.x;
 
-  if (scaled_height && m_scale.y > 0)
-    *scaled_height = texture_height * m_scale.y;
+  if (output && m_scale.y > 0)
+    output->scaled_height = input.texture_height * m_scale.y;
 }
 
-void ScaleAction::OnProjection(Common::Matrix44* matrix)
+void ScaleAction::OnProjection(const GraphicsModActionData::Input::Projection& input,
+                               GraphicsModActionData::Output::Projection* output)
 {
-  if (!matrix)
+  if (!output)
     return;
-  auto& the_matrix = *matrix;
-  the_matrix.data[0] = the_matrix.data[0] * m_scale.x;
-  the_matrix.data[5] = the_matrix.data[5] * m_scale.y;
+  output->matrix.data[0] = input.matrix.data[0] * m_scale.x;
+  output->matrix.data[5] = input.matrix.data[5] * m_scale.y;
 }
 
-void ScaleAction::OnProjectionAndTexture(Common::Matrix44* matrix)
+void ScaleAction::OnProjectionAndTexture(const GraphicsModActionData::Input::Projection& input,
+                                         GraphicsModActionData::Output::Projection* output)
 {
-  if (!matrix)
+  if (!output)
     return;
-  auto& the_matrix = *matrix;
-  the_matrix.data[0] = the_matrix.data[0] * m_scale.x;
-  the_matrix.data[5] = the_matrix.data[5] * m_scale.y;
+  output->matrix.data[0] = input.matrix.data[0] * m_scale.x;
+  output->matrix.data[5] = input.matrix.data[5] * m_scale.y;
```

Comment:
Oops, this only writes into 2 elements of the matrix, and doesn't copy the other 14!